tmol/score/omega/omega_energy_term.py

```diff
@@ -0,0 +1,88 @@
+import torch
+
+from ..atom_type_dependent_term import AtomTypeDependentTerm
+from ..bond_dependent_term import BondDependentTerm
+
+from tmol.database import ParameterDatabase
+from tmol.score.common.stack_condense import tile_subset_indices
+from tmol.score.omega.params import OmegaGlobalParams
+from tmol.score.omega.omega_whole_pose_module import OmegaWholePoseScoringModule
+
+from tmol.chemical.restypes import RefinedResidueType
+from tmol.pose.packed_block_types import PackedBlockTypes
+from tmol.pose.pose_stack import PoseStack
+
+
+class OmegaEnergyTerm(AtomTypeDependentTerm, BondDependentTerm):
+    tile_size: int = 32
+
+    def __init__(self, param_db: ParameterDatabase, device: torch.device):
+        super(OmegaEnergyTerm, self).__init__(param_db=param_db, device=device)
+
+        self.global_params = OmegaGlobalParams.from_database(
+            param_db.scoring.omega, device
+        )
+        self.tile_size = OmegaEnergyTerm.tile_size
+
+    @classmethod
+    def score_types(cls):
+        import tmol.score.terms.omega_creator
+
+        return tmol.score.terms.omega_creator.OmegaTermCreator.score_types()
+
+    def n_bodies(self):
+        return 2
+
+    def setup_block_type(self, block_type: RefinedResidueType):
+        super(OmegaEnergyTerm, self).setup_block_type(block_type)
+
+        if hasattr(block_type, "omega_quad_uaids"):
+            return
+
+        omega_quad_uaids = torch.full((4, 3), -1, dtype=torch.int32)
```

Comment:
Should I be putting these in a device torch tensor here? I think it's probably unnecessary because the setup_packed_block_types func makes a copy into a separate tensor regardless, and only that tensor is passed down into the kernel.

What is the purpose of the distinction between setup_block_type and setup_packed_block_types, then? Why not just do everything in setup_packed_block_types?

Reply:
Something that occurred to me: what do we want to do for types that don't have omega defined? Should we fill the omega_quad_uaids for that block type as all -1? That should cause the uaid resolution function to properly skip this.

Reply:
Another quick question - does it make sense to extract the omega values out here, when they are already members of RefinedResidueType? They exist in the torsion_to_uaids Mapping as a Tuple instead of a numpy type, and only exist conditionally. Not sure where to draw the line at redundancy.